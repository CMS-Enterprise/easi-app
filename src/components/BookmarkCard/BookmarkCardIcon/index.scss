.bookmark {
    &__icon {
        color: #005EA2;
        display: flex;
        cursor: pointer;
    }
}

.black {
    color: black;
}