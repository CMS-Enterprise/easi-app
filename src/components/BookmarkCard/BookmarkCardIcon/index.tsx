import React from 'react';
import classnames from 'classnames';

import './index.scss';

type BookmarkCardIconProps = {
  className?: string;
  black?: boolean;
  size: 'sm' | 'md' | 'lg';
  onClick?: React.MouseEventHandler<HTMLButtonElement>;
};

const BookmarkCardIcon = ({
  size,
  black,
  className,
  onClick
}: BookmarkCardIconProps) => {
  const classes = classnames(
    'fa',
    'fa-bookmark',
    'bookmark__icon',
    {
      'fa-1x': size === 'sm'
    },
    {
      'fa-2x': size === 'md'
    },
    {
      'fa-3x': size === 'lg'
    },
    {
      black
    },
    className
  );

  return (
    <i
      className={classes}
      onClick={onClick}
      data-testid="bookmark-icon"
      aria-hidden="true"
    />
  );
};

export default BookmarkCardIcon;
