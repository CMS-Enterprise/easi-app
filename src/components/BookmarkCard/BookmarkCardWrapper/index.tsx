import React from 'react';
import classnames from 'classnames';

type BookmarkCardWrapperProps = {
  className?: string;
  children: React.ReactNode;
};

const BookmarkCardWrapper = ({
  className,
  children
}: BookmarkCardWrapperProps) => {
  const classNames = classnames('margin-top-1 margin-bottom-1', className);
  return (
    <div className={classNames} data-testid="bookmard-card-wrapper">
      <div className="grid-row grid-gap-lg">{children}</div>
    </div>
  );
};

export default BookmarkCardWrapper;
