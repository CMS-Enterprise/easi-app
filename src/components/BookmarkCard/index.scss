.bookmark {
    display: flex;
    flex-direction: column;
    align-items: flex-start;

    &__container {        
        padding: 24px;
        width: 472px;
        height: 327px;
        left: 121px;
        top: 642px;

        /* component/card border */
        border: 2px solid #E6E6E6;
        box-sizing: border-box;
        border-radius: 4px;
    }

    &__header {
        position: static;
        width: 424px;
        height: 58px;
        left: 24px;
        top: 24px;


        /* Inside Auto Layout */

        flex: none;
        order: 0;
        align-self: stretch;
        flex-grow: 0;
        margin: 0px 0px;
    }

    &__icon {
        color: #005EA2;
    }
}