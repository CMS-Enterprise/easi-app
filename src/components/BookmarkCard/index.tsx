import React from 'react';
import { useTranslation } from 'react-i18next';
import { Button, Card, Icon } from '@trussworks/react-uswds';
import classnames from 'classnames';
import {
  GetCedarSystemIsBookmarkedDocument,
  GetCedarSystemsQuery,
  useDeleteCedarSystemBookmarkMutation
} from 'gql/generated/graphql';

import UswdsReactLink from 'components/LinkWrapper';
import { IconStatus } from 'types/iconStatus';

import './index.scss';

type BookmarkCardProps = {
  className?: string;
  statusIcon: IconStatus;
  type: 'systemProfile'; // Built in for future iterations/varations of bookmarked datasets that ingest i18n translations for headers.
};

const BookmarkCard = ({
  className,
  type,
  id,
  name,
  description,
  acronym,
  status,
  statusIcon,
  businessOwnerOrg
}: BookmarkCardProps & GetCedarSystemsQuery['cedarSystems'][number]) => {
  const { t } = useTranslation();

  const [deleteMutate] = useDeleteCedarSystemBookmarkMutation();

  const handleDeleteBookmark = (cedarSystemId: string) => {
    deleteMutate({
      variables: {
        input: {
          cedarSystemId
        }
      },
      refetchQueries: [
        {
          query: GetCedarSystemIsBookmarkedDocument,
          variables: { id: cedarSystemId }
        }
      ]
    });
  };

  return (
    <Card
      data-testid="single-bookmark-card"
      className={classnames('desktop:grid-col-6', 'grid-col-12', className)}
    >
      <div className="grid-col-12">
        <div className="bookmark__header easi-header__basic">
          <h3 className="bookmark__title margin-top-0 margin-bottom-1">
            <UswdsReactLink to={`/systems/${id}/home/top`}>
              {name}
            </UswdsReactLink>
          </h3>
          <Button
            onClick={() => handleDeleteBookmark(id)}
            type="button"
            unstyled
          >
            <Icon.Bookmark aria-hidden size={5} />
          </Button>
        </div>
        <p className="margin-0">{acronym}</p>
        <p className="bookmark__body-text line-height-body-4">{description}</p>
        <p className="margin-bottom-0">{t(`${type}:bookmark.subHeader1`)}</p>
        <p className="text-bold margin-top-1">{businessOwnerOrg}</p>
        {/*
        <Divider />
        <div className="bookmark__header easi-header__basic">
          <div>
            <p className="margin-bottom-0">
              {t(`${type}:bookmark.subHeader2`)}
            </p>
            <p className="text-bold margin-top-0 margin-bottom-0">{status}</p>
          </div>
          <SystemHealthIcon
            status={statusIcon}
            size="xl"
            className="margin-top-3"
          />
        </div>
        */}
      </div>
    </Card>
  );
};

export default BookmarkCard;
