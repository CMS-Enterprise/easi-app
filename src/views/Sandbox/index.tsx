import React, { useEffect } from 'react';

import MainContent from 'components/MainContent';

import '../SystemProfile/components/SystemDetails/index.scss';

const Sandbox = () => {
  useEffect(() => {
    document.title = 'Sandbox';
  }, []);

  return (
    <MainContent>
      <></>
    </MainContent>
  );
};

export default Sandbox;
