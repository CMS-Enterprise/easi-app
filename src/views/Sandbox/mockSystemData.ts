import { CMSOfficeAcronym } from 'constants/enums/cmsDivisionsAndOffices';
import { IconStatus } from 'types/iconStatus';

export interface SystemInfo {
  id: string;
  name: string;
  ownerName: string;
  ownerOffice: CMSOfficeAcronym;
  productionStatus: IconStatus;
  atoStatus: IconStatus;
  atoStatusText: string;
  section508Status: IconStatus;
  section508StatusText: string;
  trbStatus: IconStatus;
  trbStatusText: string;
}

export const mockSystemInfo: SystemInfo[] = [
  {
    id: '1',
    name: 'ABC123',
    ownerName: 'Jane Doe',
    ownerOffice: 'CMMI',
    productionStatus: 'success',
    atoStatus: 'success',
    atoStatusText: 'Good to go!',
    section508Status: 'success',
    section508StatusText: 'Good to go!',
    trbStatus: 'success',
    trbStatusText: 'Good to go!'
  },
  {
    id: '2',
    name: 'XYZ789',
    ownerName: 'John Doe',
    ownerOffice: 'OIT',
    productionStatus: 'warning',
    atoStatus: 'success',
    atoStatusText: 'Good to go!',
    section508Status: 'warning',
    section508StatusText: 'Not yet compliant',
    trbStatus: 'warning',
    trbStatusText: 'Needs more microservices'
  },
  {
    id: '3',
    name: '8675309',
    ownerName: 'Jennifer Doe',
    ownerOffice: 'CCIIO',
    productionStatus: 'fail',
    atoStatus: 'fail',
    atoStatusText: 'Not yet authorized',
    section508Status: 'success',
    section508StatusText: 'Good to go!',
    trbStatus: 'fail',
    trbStatusText: 'Hash and salt your passwords!'
  }
];
