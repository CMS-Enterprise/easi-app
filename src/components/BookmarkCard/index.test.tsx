import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';

import { mockSystemInfo } from 'views/Sandbox/mockSystemData';

import BookmarkCard from './index';

describe('BookmarkCard', () => {
  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <BookmarkCard {...mockSystemInfo[0]} />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders corresponding health icon for status', () => {
    const { getByTestId } = render(
      <MemoryRouter>
        <BookmarkCard {...mockSystemInfo[2]} />
      </MemoryRouter>
    );

    expect(getByTestId('system-health-icon')).toHaveClass(
      'system-health-icon-fail'
    );
  });
});
