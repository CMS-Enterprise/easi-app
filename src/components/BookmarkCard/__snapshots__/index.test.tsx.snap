// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookmarkCard matches the snapshot 1`] = `
<div
  className="tablet:grid-col-6 grid-col-12 margin-top-3"
>
  <div
    className="grid-col-12 bookmark bookmark__container"
  >
    <div
      className="bookmark__header easi-header__basic"
    >
      <h2
        className="bookmark__title margin-top-0 margin-bottom-1"
      >
        <a
          className="usa-link"
          href="/systems/1"
          onClick={[Function]}
        >
          Happiness Achievement Module
        </a>
      </h2>
      <i
        aria-hidden="true"
        className="fa fa-bookmark fa-2x bookmark__icon"
      />
    </div>
    <p
      className="margin-0"
    >
      HAM
    </p>
    <p
      className="bookmark__body-text line-height-body-4"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    Sollicitudin donec aliquam dui sed odio porta. Faucibus quam egestas
    feugiat laoreet quis. Sapien, sagittis, consectetur adipiscing elit.
    Sollicitudin donec aliquam dui sed odio porta.
    </p>
    <p
      className="margin-bottom-0"
    >
      Officer of Department
    </p>
    <p
      className="text-bold margin-top-1"
    >
      CMMI
    </p>
    <div
      className="easi-divider"
    />
    <div
      className="bookmark__header easi-header__basic"
    >
      <div>
        <p
          className="margin-bottom-0"
        >
          ATO Status
        </p>
        <p
          className="text-bold margin-top-1 margin-bottom-0"
        >
          Good to go!
        </p>
      </div>
      <i
        className="fa system-health-icon-success fa-check-circle fa-2x margin-top-2"
        data-testid="system-health-icon"
      />
    </div>
  </div>
</div>
`;
